thingsboard_gateway/connectors/bacnet/bacnet_connector.py

```diff
@@ -540,6 +543,31 @@ async def __write_property(self, address, object_id, property_id, value, priorit
             if value is None:
                 value = Null(())
 
+            if property_id == "weeklySchedule":
+                schedule = []
+                raw_schedule = literal_eval(value)
+                for idx, day in enumerate(raw_schedule):
+                    schedule.append(DailySchedule(daySchedule=[]))
+                    for sched in day:
+                        schedule[idx].daySchedule.append(
+                            TimeValue(
+                                time=time(int(sched[0].split(":")[0]), int(sched[0].split(":")[1])),
+                                value=Real(sched[1])
+                            )
+                        )
+                value = schedule
```

Comment:
Need to be moved to a separate method, smth like `__prepare_weekly_schedule_value`